backend/vehicles/src/modules/permit/application.service.ts

```diff
@@ -690,4 +719,22 @@ export class ApplicationService {
 
     return receiptNumber;
   }
+
+  async checkApplicationInProgress(originalPermitId: string): Promise<number> {
+    const count = await this.permitRepository
+      .createQueryBuilder('permit')
+      .where('permit.originalPermitId = :originalPermitId', {
+        originalPermitId: originalPermitId,
+      })
+      .andWhere('permit.permitStatus IN (:...applicationStatus)', {
+        applicationStatus: Object.values(ApplicationStatus).filter(
+          (x) =>
+            x != ApplicationStatus.CANCELLED &&
+            x != ApplicationStatus.REJECTED &&
+            x != ApplicationStatus.ISSUED,
```

Comment:
I suggest extracting this check into a helper function of some kind, so that it may be reusable and easy to modify if needed in the future, but this can be done later on